internal/consensus/state.go

```diff
@@ -524,14 +520,6 @@ func (cs *State) OnStop() {
 	// WAL is stopped in receiveRoutine.
 }
 
-// Wait waits for the the main routine to return.
-// NOTE: be sure to Stop() the event switch and drain
-// any event channels or this may deadlock
-func (cs *State) Wait() {
-	cs.evsw.Wait()
```

Comment:
Is there a reason to get rid of this wait on the event switch?

Reply:
The wait methods aren't super useful (they return roughly as soon as the stop method returns, which is triggered by the context canceling, and doesn't actually wait for things to finish, unless the service does that extra accounting. 

Looking a bit more at this, the event switch only implements noop start/stop logic, so it's not actually going to change anything in this case.

Reply:
I have a follow on PR to remove the service methods from the event switch